mail-manager/src/main/scala/it/pagopa/interop/commons/mail/InteropMailer.scala

```diff
@@ -0,0 +1,42 @@
+package it.pagopa.interop.commons.mail
+
+import courier.Defaults._
+import courier._
+import javax.mail.internet.InternetAddress
+import scala.concurrent.Future
+
+class InteropMailer private (sender: InternetAddress, mailer: Mailer) {
+
+  def send(mailData: Mail): Future[Unit] = sendMail(
+    mailData.recipients,
+    mailData.subject,
+    mailData.attachments match {
+      case Nil => Text(mailData.body)
+      case xs  =>
+        xs.foldLeft(Multipart().text(mailData.body)) { (mailContent, attachment) =>
+          mailContent.attachBytes(attachment.bytes, attachment.name, attachment.mimeType)
+        }
+    }
+  )
+
+  private def sendMail(recipients: Seq[InternetAddress], mailSubject: String, mailContent: Content) = mailer(
+    Envelope
+      .from(sender)
+      .to(recipients: _*)
+      .subject(mailSubject)
+      .content(mailContent)
+  )
+}
+
+object InteropMailer {
+  def create(): Either[Throwable, InteropMailer] = MailConfiguration.read().map { config =>
+    val mailer: Mailer = Mailer(config.smtp.serverAddress, config.smtp.serverPort)
+      .auth(config.smtp.authenticated.getOrElse(true))
+      .ssl(config.smtp.withSsl.getOrElse(true))
+      .as(config.smtp.user, config.smtp.password)
+      .startTls(config.smtp.withTls.getOrElse(true))()
+    new InteropMailer(config.sender, mailer)
+  }
+
+  def unsafeFrom(sender: InternetAddress, mailer: Mailer): InteropMailer = new InteropMailer(sender, mailer)
```

Comment:
if configurations would be passed by parameter, probably we could avoid the "unsafe" approach

Reply:
The thing is that I that for testing I need a way to pass a mailer instance, so I'll prefer to keep it

Reply:
thinking aloud: if the configurations are passed as parameters, the mailer instance can be created during the test, right?
This way we would not need a specific method just for testing

Reply:
That's the point, the Mailer instance has to be created within the test, but it's installation should be done like

```scala
  class MockedSMTPProvider
      extends Provider(Provider.Type.TRANSPORT, "mocked", classOf[MockTransport].getName, "Mock", null)

  def mockedMailer(): Mailer = {
    val properties: Properties = new Properties
    properties.put("mail.transport.protocol.rfc822", "mocked")
    val mockedSession: Session = Session.getDefaultInstance(properties)
    mockedSession.setProvider(new MockedSMTPProvider)
    Mailer(mockedSession)
  }
```

and not by passing configuration parameters. That's why these tests require that unsafe constructor